jansi/src/main/java/org/fusesource/jansi/AnsiConsole.java

```diff
@@ -100,6 +104,11 @@ public void close() throws IOException {
 		};
 	}
 
+	private static boolean isCygwin(){
+		String term = System.getenv("TERM");
+		return (term != null && term.equals("cygwin"));
```

Comment:
I'm no Cygwin expert, but I've been testing JLine2/Jansi on Cygwin for Jython 2.7. On mintty, which is the default terminal for current Cygwin, TERM is "xterm"